src/api/api.js

```diff
@@ -0,0 +1,64 @@
+const BASE_URL = `https://6.react.pages.academy/wtw`;
+const TIMEOUT = 5000;
+const STATUS_CODE = 200;
+
+class Api {
+  constructor(options = {}) {
+    this.baseUrl = BASE_URL;
+    this.timeout = options.timeout || TIMEOUT;
+  }
+
+  processResponse(response) {
+    return new Promise((resolve, reject) => {
+      if (response.status === STATUS_CODE) {
+        resolve(response.json());
+      } else {
+        reject({
+          status: response.status,
+          statusText: response.statusText
+        });
+      }
+    });
+  }
+
+  fetchWithTimeout(url, options = {}) {
+    return Promise.race([
+      fetch(url, options),
+      new Promise((_, reject) => {
+        setTimeout(() => reject(`Ожидание ответа от сервера превышено`), this.timeout);
+      }),
+    ]);
+  }
+
+  async loadFilms() {
+    return await this.fetchWithTimeout(`${this.baseUrl}/films`)
+    .then(this.processResponse)
+    .then((data) => {
```

Comment:
Если написано async то можно (и нужно) без then 
```suggestion
    this.processResponse)
    .then((data) => {
```